### ISSUE TITLE
Spike: Find out a unified async-to-sync pattern

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#18]
### Expected result

For Xperience APIs that aren't async yet, find a way to use them in the live site's async code base safely and efficiently.

### Additional context

Add any other context or guidelines here.

### Resources

* URL

### Issue expands upon

This issue expands upon another issue(s): #`issue number`

### Blockers

This issue is blocked by the following issue(s): #`issue number`
